### PR TITLE
Update the iot-core demo to use Broker rather than Channel

### LINKED ISSUE #1021: Eventing sample updates for v0.5
[/docs/docs/eventing/samples](https://github.com/knative/docs/tree/master/docs/eventing/samples)

- [x] gcp-pubsub-source
- [x] iot-core
- [x] kubernetes-event-source

### PR DESCRIPTION
**Pull request by Harwayne**
Part of #1021

## Proposed Changes

- Update the iot-core demo to use Broker rather than Channel
- Removes `ko` and `kail` usage.

### CODE REVIEW

**Comment by Harwayne** [comment]
Ping.

File: docs/eventing/samples/iot-core/README.md [hunk 1]
```diff
@@ -199,37 +173,39 @@ see them in the subscriber.
...
+    
+    ```shell
+    {"ID":"481014114648052","Data":"eyJzb3VyY2VfaWQiOiJpb3QtY29yZSBkZW1vIiwiZXZlbnRfaWQiOiJlaWQtMzI3MjJiMzItZWU5Mi00YzZlLWEzOTgtNDlmYjRkYWYyNGE1IiwiZXZlbnRfdHMiOjE1NTM3MTczOTYsIm1ldHJpYyI6MC4xMzY1MjI5OH0=","Attributes":{"deviceId":"iot-demo-client","deviceNumId":"2754785852315736","deviceRegistryId":"iot-demo","deviceRegistryLocation":"us-central1","projectId":"s9-demo","subFolder":""},"PublishTime":"2019-03-27T20:09:56.685Z"}
+
 ### Cleanup
 
 To cleanup the knative resources:
 
-1.  Remove the gcppubsub controller:
+1.  Remove the `GcpPubSubSource`:
```
Comment (samodell) [inline docs/eventing/samples/iot-core/README.md]: Do users need to replace anything in this command before running it? If so, it's good to call out what needs to be replaced with the user's data.

Comment (Harwayne) [reply docs/eventing/samples/iot-core/README.md]: No, the command is self-contained. It requires no changes to the YAML ahead of time. 

It does require that the user runs the command in a shell with the proper environment variables defined. See the "Environment Variables" section.

**Pull request merged**